src/stats-watcher.ts

```diff
@@ -0,0 +1,305 @@
+// ============================================================================
+// AUTHPROXY - STATS WATCHER MODULE
+// Monitors game database for match completions and triggers webhooks
+// ============================================================================
+
+import { PrismaClient } from '@prisma/client';
+import Logger from './logger';
+import { TrophyWebhook } from './webhooks-trophy';
+import { MetArenaWebhook } from './webhooks-metarena';
+
+const log = new Logger();
+
+export interface StatsWatcherConfig {
+    enabled: boolean;
+    gameDbUrl: string;
+    pollIntervalMs: number;
+}
+
+interface PlayerStats {
+    wins: number;
+    mvps: number;
+    total_games_played: number;
+    current_mmr: number;
+    win_streak: number;
+}
+
+interface CachedPlayer {
+    stats: PlayerStats;
+    odinalId: string;
+    username: string;
+    publisherUsername: string;
+    playlistGuid: string;
+    matchFlow: number;
+    lastSeen: Date;
+}
+
+export class StatsWatcher {
+    private config: StatsWatcherConfig;
+    private gamePrisma: PrismaClient | null = null;
+    private playerCache: Map<string, CachedPlayer> = new Map();
+    private pollInterval: NodeJS.Timeout | null = null;
+    private isRunning: boolean = false;
+    private trophyWebhook: TrophyWebhook | null;
+    private metarenaWebhook: MetArenaWebhook | null;
+
+    constructor(
+        config: StatsWatcherConfig,
+        trophyWebhook: TrophyWebhook | null = null,
+        metarenaWebhook: MetArenaWebhook | null = null
+    ) {
+        this.config = config;
+        this.trophyWebhook = trophyWebhook;
+        this.metarenaWebhook = metarenaWebhook;
+    }
+
+    async start(): Promise<boolean> {
+        if (!this.config.enabled) {
+            log.info('[StatsWatcher] Disabled in config');
+            return false;
+        }
+
+        if (this.isRunning) {
+            log.warn('[StatsWatcher] Already running');
+            return true;
+        }
+
+        log.info('[StatsWatcher] Starting...');
+        log.info(`[StatsWatcher] Game DB: ${this.config.gameDbUrl.replace(/:[^:@]+@/, ':****@')}`);
+        log.info(`[StatsWatcher] Poll interval: ${this.config.pollIntervalMs}ms`);
+        log.info(`[StatsWatcher] Trophy webhook: ${this.trophyWebhook ? 'enabled' : 'disabled'}`);
+        log.info(`[StatsWatcher] MetArena webhook: ${this.metarenaWebhook ? 'enabled' : 'disabled'}`);
+
+        try {
+            this.gamePrisma = new PrismaClient({
+                datasources: {
+                    db: { url: this.config.gameDbUrl }
+                }
+            });
+            await this.gamePrisma.$connect();
```

Comment:
Why do we need a second database connection? You could simply use the global prisma client we already have.